libear/ear.c

```diff
@@ -450,7 +451,7 @@ static void bear_report_call(char const *const argv[]) {
         perror("bear: mkstemp");
         exit(EXIT_FAILURE);
     }
-    if (0 > bear_write_json_report(fd, argv, cwd, getpid())) {
+    if (0 > bear_write_json_report(fd, argv, cwd, getpid()) && errno) {
```

Comment:
hmm, thinking back, maybe it would be better to fix whatever encoding problem `bear_write_json_report` is having, rather than any of these changes.  That way we can still fail hard, rather than paper over any issues (the approach this version takes).